docs/docs/uploads.md

```diff
@@ -480,11 +480,11 @@ const filesWhereQuery = await db.file.findMany({
 
 // 🛑 Will not work, because files accessed via relation
 // highlight-next-line
-return filesViaRelation.map((file) => file.withSignedUrl())
+return filesViaRelation.map(async (file) => await file.withSignedUrl())
```

Comment:
Does having `async` + `await` here make a difference? I thing you'll be returning an array of promises either way, but here you're creating an extra layer of promise wrapping (which the JIT compiler might be smart enough to remove, idk)

Reply:
I don't know enough about promises to say one way or the other. I know that I need the `withSignedUrl` to be async so that the s3 sdk can call:

```
  async sign(fileLocation: string, expiresIn = 3600): Promise<string> {
    const command = new GetObjectCommand({
      Bucket: this.bucket,
      Key: fileLocation,
    })

    return await getSignedUrl(this.s3Client, command, { expiresIn })
  }
```

Without it didn't work.

Reply:
Yeah, I totally see why you had to make it async. No question there 🙂 Just don't think this need the extra async/await. But please don't just take my word for it. Try it out in a real project to make sure.

Reply:
I did. Without the await in my describe pic and remove background job when I asked for the signedUrl it was undefined. When I made it async I got the url. Maybe within the service you do r need it but in a job I did so I could get the url and pass it along to fal as an image url reference.

Reply:
Also the dataUri compute helper was always async so I just made both compute helpers work the same.

Reply:
Yes, that all makes perfect sense. When you need to use the value for further computations/logic you need to await it first. But in the example you're just straight away returning it. That's why I don't think you need to await it. Just just passing the promise along.

This is the current example:
```ts
return filesViaRelation.map(async (file) => await file.withSignedUrl())
```

I'm just saying that in that particular case I think you could just have done this
```ts
return filesViaRelation.map((file) => file.withSignedUrl())
```
Because `withSignedUrl()` returns a promise and that promise is then used as the return value of the `map` callback. 

If you had to use `file` inside the callback after calling `withSignedUrl()` you would have had to await it first. Like this:
```ts
return filesViaRelation.map(async (file) => {
  const fileWithSignedUrl = await file.withSignedUrl()
  console.log('fileWithSignedUrl', fileWithSignedUrl)

  return fileWithSignedUrl
})
```

So yes, `withSignedUrl()` always has to be async. But whether you explicitly need to await it or not depends on how you use it.

I took an extra look inside `S3Storage.ts` and now see that I have similar feedback there. I think we should directly return the promise in `remove` and `sign` instead of first awaiting it and then returning.

But that requires some changes to `BaseStorageAdapter`. So maybe hold off on these changes until after we've discussed our current implementation on Tuesday